File: BoardGamesNook.Model/BoardGame.cs

using System;

namespace BoardGamesNook.Model
{
    public class BoardGame
    {
        public int Id { get; set; }
        public DateTimeOffset CreatedDate { get; set; }
        public DateTimeOffset ModifiedDate { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public int MinPlayers { get; set; }
        public int MaxPlayers { get; set; }
        public int MinAge { get; set; }
        public int MinTime { get; set; }
        public int MaxTime { get; set; }
        public int? BGGId { get; set; }
        public bool IsExpansion { get; set; }
        public int? ParentId { get; set; }
        public bool IsConfirmed { get; set; }
        public bool Active { get; set; }
        public string ImageUrl { get; set; }

    public virtual BoardGame ParentBoardGame { get; set; }
    }
}

File: BoardGamesNook/src/boardGames/boardGame.ts
export class BoardGame {
    Id: number;
    Name: string;
    Description: string;
    MinPlayers: number;
    MaxPlayers: number;
    MinAge: number;
    MinTime: TimeRanges;
    MaxTime: TimeRanges;
    BGGId: number | null;
    IsExpansion: boolean;
    ParentId: number | null;
    ParentBoardGame: BoardGame | null;
    ImageUrl: string;
}